plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/JoinOperator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+public enum JoinOperator
+{
+    JOIN("JOIN"),
+    LEFT_JOIN("LEFT JOIN"),
+    RIGHT_JOIN("RIGHT JOIN"),
+    FULL_JOIN("FULL JOIN"),
```

Comment:
I think this operator it's more about how JOIN condition is present in actual String query,
for example in future or for some connectors we could write LEFT OUTER JOIN instead of LEFT JOIN.
So I'd prefer to keep these things separately

Reply:
I don't know how to tackle already defined enums and their string representation elegantly. 
And this does not seem to change frequently or any time soon. 
so may be at least some verify (JoinOperator.values.size() vs JoinType.values().size())?
> or for some connectors (in case of some bugs) we could add/write LEFT OUTER JOIN instead/additionally to LEFT JOIN.

don't see how to do that easily with current implementation, but I think it does not matter here/now.

Reply:
> don't see how to do that easily with current implementation, but I think it does not matter here/now.

you can just add to enum : 
...
LEFT_OUTER_JOIN("LEFT OUTER JOIN"),
...

Reply:
> so may be at least some verify (JoinOperator.values.size() vs JoinType.values().size())?

For me it's not mapping to actual joins, maybe another naming will help - like JoinOperatorStringRepresentation, don't know

Reply:
It's important to decouple the following three things:

1. Join types on the plan level (represented by `JoinNode.Type`)
2. Join types on SPI level (represented by `JoinType`)
3. Mapping SPI join types to SQL strings (represented by `JoinOperators` here).

There's no reason for 1:1 mapping between 1 and 2.
There also no reason for 1:1 mapping between 2 and 3. e.g. `LEFT OUTER` can appear in SQL text as `LEFT` or `LEFT OUTER` - both are same thing.

Reply:
It would be useful to verify `JoinType` values are subset of `JoinOperator` but it sounds premature - it's only a problem when someone implements a new Join node, adds plan optimizer to push down to table scan, implements in some connector - all of this without adding tests.